09-redux/classworks/Gloria_Loi/src/resources/actions/fetchSubreddits.js

```diff
@@ -0,0 +1,32 @@
+import { fetchPostsArticles } from './fetchPostArticles';
+
+export const fetchSubreddits = (inputString, fetchReddit, history) => async (
+  dispatch
+) => {
+  if (inputString === '') {
+    fetchPostsArticles(undefined, fetchReddit, history)(dispatch);
+    return;
+  }
+
+  const data = await fetchReddit(
+    `/subreddits/search?limit=10&q=${inputString}`
+  ).then((res) => res.json());
+
+  dispatch({
+    type: 'SET_SUBREDDITS',
+    data,
+  });
+
+  if ((data.data.children && data.data.children.length === 0) || !data) {
```

Comment:
data.data.data.data.data :thinking: 
better rename your data object to response or something

Reply:
btw this check fill break the app if there will no data, as you check `data.data.children` and only after that you check `!data`